lib/ui/painting/scene/scene_shader.cc

```diff
@@ -0,0 +1,82 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/painting/scene/scene_shader.h"
+
+#include <memory>
+#include <utility>
+
+#include "flutter/display_list/display_list_color_source.h"
+#include "flutter/display_list/display_list_tile_mode.h"
+#include "flutter/lib/ui/dart_wrapper.h"
+#include "flutter/lib/ui/painting/scene/scene_node.h"
+#include "flutter/lib/ui/ui_dart_state.h"
+#include "impeller/geometry/scalar.h"
+#include "impeller/geometry/size.h"
+
+#include "third_party/tonic/converter/dart_converter.h"
+#include "third_party/tonic/dart_args.h"
+#include "third_party/tonic/dart_binding_macros.h"
+#include "third_party/tonic/dart_library_natives.h"
+#include "third_party/tonic/typed_data/typed_list.h"
+
+namespace flutter {
+
+IMPLEMENT_WRAPPERTYPEINFO(ui, SceneShader);
+
+SceneShader::SceneShader(fml::RefPtr<SceneNode> scene_node)
+    : scene_node_(std::move(scene_node)) {}
+
+void SceneShader::Create(Dart_Handle wrapper, Dart_Handle scene_node_handle) {
+  auto* scene_node =
+      tonic::DartConverter<SceneNode*>::FromDart(scene_node_handle);
+  if (!scene_node) {
+    return;
+  }
+
+  auto res = fml::MakeRefCounted<SceneShader>(fml::Ref(scene_node));
+  res->AssociateWithDartWrapper(wrapper);
+}
+
+void SceneShader::SetCameraTransform(const tonic::Float64List& matrix4) {
+  camera_transform_ =
+      impeller::Matrix(matrix4[0], matrix4[1], matrix4[2], matrix4[3],    //
+                       matrix4[4], matrix4[5], matrix4[6], matrix4[7],    //
+                       matrix4[8], matrix4[9], matrix4[10], matrix4[11],  //
+                       matrix4[12], matrix4[13], matrix4[14], matrix4[15]);
+}
+
+static impeller::Matrix DefaultCameraTransform() {
+  // TODO(bdero): There's not way to know what the draw area will be yet, so
```

Comment:
"There's no way". Here and elsewhere, perhaps file bugs. But if this is for a demo, we can do a pass later to rethink this.